.github/copilot-instructions.md

```diff
@@ -0,0 +1,293 @@
+└── *.md                     # Various documentation files
+```
+
+### Environment Variables for Development
+Set these for debugging and development:
+- `MSBUILDDEBUGONSTART=2` - Wait for debugger attach on MSBuild start
+- `MSBUILDLOGVERBOSERARSEARCHRESULTS=1` - Verbose ResolveAssemblyReference logging
+- `MSBUILDDISABLENODEREUSE=1` - Disable MSBuild process reuse
+
+### Debugging Options
+For debugging MSBuild:
+- Set `MSBUILDDEBUGONSTART=2` environment variable, then attach a debugger after the process starts
+- Use Visual Studio with "Use previews of the .NET SDK (requires restart)" enabled
+- Build first, then run `artifacts\msbuild-build-env.bat` (Windows) or `source artifacts/sdk-build-env.sh` (Linux/macOS) before debugging
+
+### Contributing Guidelines
+Before submitting code:
+- Discuss significant changes with the team first
+- Follow [.NET Runtime Coding Guidelines](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md)
+- Use Shouldly assertion framework for new or modified test cases
+- Ensure backwards compatibility
+- Write tests for new code paths and bug fixes
+- Performance is critical: minimize allocations, avoid LINQ, use efficient algorithms
+
+## Troubleshooting
+
+### Common Issues and Solutions
+
+**Build fails with "Could not resolve SDK":**
+- Ensure you run `source artifacts/sdk-build-env.sh` after building
+- Verify `dotnet --version` shows the preview version (10.0.100-preview.7.25372.107)
+
+**Tests fail:**
+- This is EXPECTED for some tests in CI/development environments
+- Run individual test projects instead: `dotnet test src/Framework.UnitTests/Microsoft.Build.Framework.UnitTests.csproj`
+
+**Certificate errors with external projects:**
+- Use the repository's sample projects for testing instead of creating new external projects
+- The bootstrap environment is designed for building the MSBuild repository itself
+
+**Network connectivity issues during build:**
+- "Resource temporarily unavailable" errors from dnceng.pkgs.visualstudio.com indicate Azure DevOps connectivity issues
+- These are environment/network related, not code issues
+- Retry the build or work in an environment with better connectivity to Microsoft's package feeds
+
+**Long build times:**
+- First build after clone takes longer due to package restoration
+- Subsequent builds are faster (incremental)
+- NEVER cancel long-running commands - they WILL complete
+
+### Files You Should NOT Modify
+- `global.json` - Controls .NET SDK version
+- `NuGet.config` - Package source configuration
+- `artifacts/` directory contents - Generated during build
+- `.dotnet/` directory contents - Bootstrap SDK location
+
+### Key Files to Understand
+- `build.sh` / `build.cmd` - Main build entry points
+- `eng/common/build.sh` - Actual build implementation  
+- `MSBuild.sln` - Main solution file (large, contains all projects)
+- `MSBuild.Dev.slnf` - Filtered solution for development
```

Comment:
Does anybody use the `.slnf` these days? I've been back on the full solution for years now.